tuf/ngclient/updater.py

```diff
@@ -292,8 +293,15 @@ def _load_local_metadata(self, rolename: str) -> bytes:
             return f.read()
 
     def _persist_metadata(self, rolename: str, data: bytes):
-        with open(os.path.join(self._dir, f"{rolename}.json"), "wb") as f:
-            f.write(data)
+        """To ensure there is no chance for loss of data during writing, the data is first written to a
+        temp file followed by an atomic move operation to the correct file location to make sure that 
+        if data writing process is halted or interrupted, the original data is not lost.
+        """
+        
+        original_filename = open(f"{rolename}.json", 'a+')
```

Comment:
I'm not sure we support f-strings yet. Maybe best to stick to our current convention... which is String.format()? @jku

Reply:
The original code that I pulled my first fork out of used `f-strings` in place, that is why I thought this would be the correct way to go about it. 
When I looked at the tests and where they failed, it seems my mistake is the fact that I am passing an `iowrapper` to the `NamedTemporaryFile` function where it expects a `str`, so I think that change could possibly solve the thing. 
But I really do not understand why is it that these errors do not come up when I run `tox` or the `aggregate_tests.py` functions on my system.